src/runner.rs

```diff
@@ -1252,65 +1108,92 @@ pub fn check_function_args(signature: &Signature, args: &[RuntimeValue]) -> Resu
 	Ok(())
 }
 
+#[derive(Debug)]
 struct ValueStack {
-	stack_with_limit: StackWithLimit<RuntimeValue>,
+	buf: Box<[RuntimeValue]>,
+	/// Index of the first free place in the stack.
+	sp: usize,
 }
 
 impl ValueStack {
 	fn with_limit(limit: usize) -> ValueStack {
+		let mut buf = Vec::new();
+		buf.resize(limit, RuntimeValue::I32(0));
```

Comment:
If we unify the value types as in #99 we can make this use `calloc`

Reply:
Yeah, that's the plan! (And because of this I've asked you about zero-ing FPs the other day : ) )

Reply:
Great! I don't know whether it will work with a union but we can do `Vec::<u64>::new()` and then convert that to a `Vec<RuntimeValue>`. That'll definitely use `calloc`.